frontend/apps/reader/modules/readerdictionary.lua

```diff
@@ -313,14 +304,50 @@ If you'd like to change the order in which dictionaries are queried (and their r
 end
 
 function ReaderDictionary:onLookupWord(word, box, highlight, link)
+    logger.dbg("lookup word:", word, box)
+    -- escape quotes and other funny characters in word
+    word = self:cleanSelection(word)
+    logger.dbg("stripped word:", word)
+
     self.highlight = highlight
+
     -- Wrapped through Trapper, as we may be using Trapper:dismissablePopen() in it
     Trapper:wrap(function()
-        self:stardictLookup(word, box, link)
+        self:stardictLookup(word, self.enabled_dict_names, not self.disable_fuzzy_search, box, link)
     end)
     return true
 end
 
+function ReaderDictionary:onHtmlDictionaryLinkTapped(dictionary, link)
+    if not link.uri then
+        return
+    end
+
+    -- The protocol is either "bword" or there is no protocol, only the word.
+    -- https://github.com/koreader/koreader/issues/3588#issuecomment-357088125
+    local word = link.uri
```

Comment:
Would it be clearer to write it like

```lua
local word -- equivalent to local word = nil
if prefix then word = link.uri:sub
elseif :// then return
else word = link.uri end
```

(Pay no mind to the formatting.)

Reply:
To my non-Lua trained eyes those single line ifs are a bit harder to read but I've changed the code.

Reply:
ah no no no! D-:

I meant the if elseif else structure. Hence "don't mind the formatting" I'm just typing this up in 10 seconds. :-P

Reply:
So do you meant this? :)
https://github.com/TnS-hun/koreader/blob/patch-html-dictionary-link-support/frontend/apps/reader/modules/readerdictionary.lua#L328-L336

Reply:
Yes, exactly. 👍